bootstrap/secret/insecure.go

```diff
@@ -146,3 +148,40 @@ func (p *InsecureProvider) ListSecretPaths() ([]string, error) {
 
 	return results, nil
 }
+
+// RegisteredSecretUpdatedCallback registers a callback for a secret.
+func (p *InsecureProvider) RegisteredSecretUpdatedCallback(path string, callback func(path string)) (func(), error) {
+	if _, ok := p.registeredSecretCallbacks[path]; ok {
+		return func() {
+			p.DeregisterSecretUpdatedCallback(path)
+		}, fmt.Errorf("there is a callback already registered for path '%v'", path)
+	}
+
+	// Register new call back for path.
+	p.registeredSecretCallbacks[path] = callback
+
+	return func() {
+		p.DeregisterSecretUpdatedCallback(path)
+	}, nil
+}
```

Comment:
```suggestion
func (p *InsecureProvider) RegisteredSecretUpdatedCallback(path string, callback func(path string)) (func(), error) {
	if _, ok := p.registeredSecretCallbacks[path]; ok {
		return nil, fmt.Errorf("there is a callback already registered for path '%s'", path)
	}

	// Register new call back for path.
	p.registeredSecretCallbacks[path] = callback

	return func() {
		if _, ok := p.registeredSecretCallbacks[path]; ok {
			delete(p.registeredSecretCallbacks, path)
		}
	}, nil
}
```

Reply:
1. If there is an error, don't need to return the func()
2. Inline the code to deregister the function.  Don't need a public Deregister.